test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -612,4 +610,28 @@ protected final RestClientBuilder getClientBuilderWithSniffedHosts() throws IOEx
     public ClientYamlTestCandidate getTestCandidate() {
         return testCandidate;
     }
+
+    private static class SkipSectionContextAdapter implements SkipSectionContext {
```

Comment:
I'm wondering, is `SkipSectionContext` really necessary? Imho things are becoming just harder to understand due to the added level of indirection. As far as I can see this is only beneficial for testing. Wouldn't it be possible to just mock `ClientYamlTestExecutionContext` to achieve the same and use `skip(ClientYamlTestExecutionContext)` instead?

Reply:
I prefer (and find easier to understand) when things are more scoped and dependencies have clear cut boundaries, and then add adapters when needed. Besides making this easier to test (as you noticed), it makes it clearer what Skip (SkipSection/SkipCriteria) uses and what not. 
That said, since these classes are all specific to section execution in Yaml tests, I see not much as a problem to get rid of SkipSectionContext and use ClientYamlTestExecutionContext directly. I can go in that direction.

Reply:
> it makes it clearer what Skip (SkipSection/SkipCriteria) uses and what not

Absolutely on board with that part, but in that case I would prefer if SkipSectionContext was just a subset of the broader context (os, clusterHasFeature, esVersion). You're still implementing your skip logic on the full scope of ClientYamlTestExecutionContext, and the tricky / twisted bit is that the checks are not consistent with respect to skipping (clusterIsRunningOs / clusterVersionInRange vs clusterHasFeature). But consider it a nit.

Reply:
Yes, I agree they are identical now (same set of things exposed) but with different logic. Since there is very little utility for `SkipSectionContext` as is, I will remove it.